### PR TITLE
Bump to version 0.5.0

### PR DESCRIPTION
**Pull request by neerajprad**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -84,7 +84,7 @@
         # numpy is necessary for some functionality of PyTorch
         'numpy>=1.7',
         'opt_einsum>=2.3.2',
-        'pyro-api@https://api.github.com/repos/pyro-ppl/pyro-api/tarball/master',
+        'pyro-api@0.1.0',
```
Comment (fritzo) [inline setup.py]: `pyro-api>=0.1.0`

File: .travis.yml [hunk 2]
```diff
@@ -24,6 +24,9 @@ cache:
 
 install:
     - pip install -U pip
+    - pip install torch==1.2.0+cpu torchvision==0.4.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
+    # Keep track of pyro-api master branch
+    - pip install https://github.com/pyro-ppl/pyro-api/archive/master.zip
     - pip install torch==1.3.0+cpu torchvision==0.4.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
```
Comment (jpchen) [inline .travis.yml]: is there a reason you're installing torch 1.2, then 1.3 ?

Comment (neerajprad) [reply .travis.yml]: 🤦‍♂ That's due to a bad merge. Thanks for catching!

**Comment by neerajprad** [comment]
The sparse regression example seems to be failing here too. Changing `num-dimensions` to 20 seems to fix this. @martinjankowiak - this seems to fail due to some 0 sized dims in `active_quadratic_dims`.  

```
python examples/sparse_regression.py --num-steps=2 --num-data=50 --num-dimensions 30
Beginning MAP optimization...
[step 0000]  loss: 8.67906
[step 0001]  loss: 6.10070
Expected singleton thetas:
 [ 0.29434243  0.358192   -0.02293254 -0.14172214  0.01926203  0.06307288
 -0.35329664 -0.02380897 -0.13799758  0.25468335]
Identified the following active dimensions: [0 1]
Mean estimate for active singleton weights:
 [0.1048793  0.13842413]
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/numpy/lib/shape_base.py", line 843, in split
    len(indices_or_sections)
TypeError: object of type 'int' has no len()
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/travis/build/pyro-ppl/pyro/examples/sparse_regression.py", line 321, in <module>
    main(args)
  File "/home/travis/build/pyro-ppl/pyro/examples/sparse_regression.py", line 285, in main
    pyro.param('auto_var_obs').double())
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/torch/autograd/grad_mode.py", line 49, in decorate_no_grad
    return func(*args, **kwargs)
  File "/home/travis/build/pyro-ppl/pyro/examples/sparse_regression.py", line 172, in compute_posterior_stats
    active_quadratic_dims = np.split(active_quadratic_dims, active_quadratic_dims.shape[0])
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/numpy/lib/shape_base.py", line 847, in split
    if N % sections:
ZeroDivisionError: integer division or modulo by zero
FAILED
```

**Comment by neerajprad** [comment]
Ready to merge.

**Pull request merged**